src/main/java/org/killbill/billing/plugin/api/PluginProperties.java

```diff
@@ -88,7 +89,7 @@ public boolean apply(final PluginProperty input) {
             return null;
         }
         final String pluginPropertyString = String.valueOf(pluginProperty.getValue());
-        return Strings.isNullOrEmpty(pluginPropertyString) ? null : pluginPropertyString;
+        return Strings.isNullOrEmpty(pluginPropertyString) ? null : UTF8UrlDecoder.decode(pluginPropertyString).toString();
```

Comment:
I don't think we should change the current behavior of `findPluginPropertyValue` (it would change the behavior of all plugins relying on it). Instead, let's introduce a new method (e.g. `findAndDecodePluginPropertyValue`).